src/components/AssetSearch/selectors/selectAndSortAssets/selectAndSortAssets.ts

```diff
@@ -1,18 +1,37 @@
+import { CoinGeckoMarketCapResult } from '@shapeshiftoss/types'
 import sortBy from 'lodash/sortBy'
 import { createSelector } from 'reselect'
 import { ReduxState } from 'state/reducer'
 import { AssetsState, FullAsset } from 'state/slices/assetsSlice/assetsSlice'
 
 export const selectAndSortAssets = createSelector(
   (state: ReduxState) => state.assets,
-  (assets: AssetsState) => {
+  (state: ReduxState) => state.marketData.marketCap,
+  (assets: AssetsState, marketCap?: CoinGeckoMarketCapResult) => {
     let sortedAssets: FullAsset[] = []
     const assetsEntries = Object.entries(assets)
-    if (assetsEntries.length > 0) {
-      const data = assetsEntries.map(([, val]) => {
-        return val
-      })
-      sortedAssets = sortBy(data, ['name', 'symbol'])
+    if (marketCap) {
+      // we only fetch market data for the top 250 assets
+      // and want this to be fairly performant so do some mutatey things
+      const assetsByCAIP19 = assetsEntries.reduce<Record<string, FullAsset>>((acc, [, cur]) => {
+        acc[cur.caip19] = cur
+        return acc
+      }, {})
+      const caip19ByMarketCap = Object.keys(marketCap)
+      const sortedWithMarketCap = caip19ByMarketCap.reduce<FullAsset[]>((acc, cur) => {
+        acc.push(assetsByCAIP19[cur])
```

Comment:
any potential for assetsByCAIP19 to return null/undefined?

Reply:
highly unlikely coingecko doesn't have asset data for the top 250 by market cap, but i'll add a check for this if some coin pricing is temporarily unavailable

Reply:
done